mmdet/apis/train.py

```diff
@@ -68,6 +68,49 @@ def set_random_seed(seed, deterministic=False):
         torch.backends.cudnn.benchmark = False
 
 
+def scale_lr(cfg, logger):
+    """Automatically scaling LR according to GPU number and sample per GPU.
+
+    Args:
+        cfg (config): training config.
+        logger (logging.Logger): logger.
+    """
+
+    if cfg.enable_auto_scale_lr is False:
+        logger.info(f'You disabled automatically scaling LR.')
+        return
+
+    if "mmdet_official_special_batch_size" in cfg:
+        original_batch_size = cfg.mmdet_official_special_batch_size
+    else:
+        original_batch_size = cfg.default_batch_size
+
+    gpu_number = len(cfg.gpu_ids)  # get the gpu number
+    batch_size = gpu_number * cfg.data.samples_per_gpu  # calculate the batch size
+
+    if batch_size == original_batch_size:
+        logger.info(f'You are using {gpu_number} GPU(s) ,'
+                    f'and {cfg.data.samples_per_gpu} samples per gpu, '
+                    f'batch size is {batch_size} which '
+                    f'match the original batch size: {original_batch_size}, '
+                    f'won\'t scaling the LR ({cfg.optimizer.get("lr")}).')
+    else:
+        # Get original LR from config
+        original_lr = cfg.optimizer.get("lr", 0)
+        assert original_lr != 0
+
+        # scale LR according to paper [linear scaling rule](https://arxiv.org/abs/1706.02677)
+        scaled_lr = (batch_size / original_batch_size) * original_lr
+        cfg.optimizer.update({"lr": scaled_lr})
+
+        logger.info(f'You are using {gpu_number} GPU(s) '
+                    f'and {cfg.data.samples_per_gpu} samples per gpu '
+                    f'means the batch is {batch_size}, '
+                    f'while mmdet original batch size of this model '
+                    f'is {original_batch_size}, '
+                    f'automatically scaling LR from {original_lr} to {scaled_lr}')
```

Comment:
When `batch_size == original_batch_size` the scaling ratio is 1. We can merge the if branch.

Reply:
Done